patchwork-recipes/src/main/java/net/patchworkmc/mixin/recipes/MixinRecipeGridAligner.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.patchworkmc.mixin.recipes;
+
+import java.util.Iterator;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+import net.minecraftforge.common.crafting.IShapedRecipe;
+
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.recipe.Recipe;
+import net.minecraft.recipe.RecipeGridAligner;
+
+@Mixin(RecipeGridAligner.class)
+public interface MixinRecipeGridAligner {
+	@Shadow
+	void acceptAlignedInput(Iterator<?> inputs, int slot, int amount, int gridX, int gridY);
+
+	/**
+	 * @author qouteall
+	 * @reason https://github.com/FabricMC/Mixin/issues/15
+	 */
+	@Overwrite
+	default void alignRecipeToGrid(
+			int gridWidth, int gridHeight, int gridOutputSlot,
+			Recipe<?> recipe, Iterator<?> inputs, int amount
+	) {
+		int width = gridWidth;
+		int height = gridHeight;
+
+		if (recipe instanceof IShapedRecipe) {
```

Comment:
Can we get some markers on where the actual code changes are with comments? I'm assuming it's just here but it's difficult to tell.

Reply:
Yes, it's generally good practice to annotate the changed parts when using `@Overwrite`.